internal/clients/composer/client.go

```diff
@@ -73,7 +77,76 @@ func createClient(composerURL string, ca string) (*http.Client, error) {
 	return &http.Client{Transport: transport}, nil
 }
 
+func shouldRetry(err error) bool {
+	if e, ok := err.(net.Error); ok {
+		// "e.Temporary" is deprecated so let's use e.Timeout
+		// as an approximation. We may need to handle ECONNRESET
+		// or similar but let's do as we go
+		// https://github.com/golang/go/issues/45729
+		return e.Timeout()
+	}
+	return false
+}
+
+var errRetryFromStatusCode = fmt.Errorf("retry based on http status code")
+
 func (cc *ComposerClient) request(method, url string, headers map[string]string, body io.ReadSeeker) (*http.Response, error) {
+	retryableHttpCodes := []int{http.StatusServiceUnavailable}
+
+	var resp *http.Response
```

Comment:
this needs tests, probably very similar to the cases in https://github.com/osbuild/image-builder/pull/1268/files#diff-6850c3f59702d7b54878bb8eac65ad4a329f1332df31329d8623521f93326f8dR253